main.go

```diff
@@ -5,58 +5,145 @@ import (
+			Short:    "sends metrics to a wavefront proxy using the wavefront data format",
 			Timeout:  10,
-			Keyspace: "sensu.io/plugins/{{ .Name }}/config",
+			Keyspace: "sensu.io/plugins/sensu-wavefront-handler/config",
 		},
 	}
 
-	handlerConfigOptions = ConfigOptions{
-		Example: sensu.PluginConfigOption{
-			Path:      "example",
-			Env:       "HANDLER_EXAMPLE",
-			Argument:  "example",
-			Shorthand: "e",
+	opts = []*sensu.PluginConfigOption{
+		&sensu.PluginConfigOption{
+			Path:      host,
+			Env:       "WAVEFRONT_HOST",
+			Argument:  host,
+			Shorthand: "",
+			Default:   "127.0.0.1",
+			Usage:     "the host of the wavefront proxy",
+			Value:     &handlerConfig.Host,
+		},
+		&sensu.PluginConfigOption{
+			Path:      port,
+			Env:       "WAVEFRONT_METRICS_PORT",
+			Argument:  port,
+			Shorthand: "m",
+			Default:   2878,
+			Usage:     "the port of the wavefront proxy",
+			Value:     &handlerConfig.MetricsPort,
+		},
+		&sensu.PluginConfigOption{
+			Path:      flush,
+			Env:       "WAVEFRONT_FLUSH_INTERVAL_SECONDS",
+			Argument:  flush,
+			Shorthand: "f",
+			Default:   1,
+			Usage:     "the flush interval of the wavefront proxy (in seconds)",
+			Value:     &handlerConfig.FlushIntervalSeconds,
+		},
+		&sensu.PluginConfigOption{
+			Path:      prefix,
+			Env:       "WAVEFRONT_PREFIX",
+			Argument:  prefix,
+			Shorthand: "p",
 			Default:   "",
-			Usage:     "An example configuration option",
-			Value:     &handlerConfig.Example,
+			Usage:     "the prefix to append to the metric name",
+			Value:     &handlerConfig.Prefix,
+		},
+		&sensu.PluginConfigOption{
+			Path:      tags,
+			Env:       "WAVEFRONT_TAGS",
+			Argument:  tags,
+			Shorthand: "t",
+			Default:   nil,
+			Usage:     "the additional tags to merge with the metric tags",
+			Value:     &handlerConfig.Tags,
 		},
-	}
-
-	options = []*sensu.PluginConfigOption{
-		&handlerConfigOptions.Example,
 	}
 )
 
 func main() {
-	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, options, checkArgs, executeHandler)
+	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, opts, checkArgs, executeHandler)
 	handler.Execute()
 }
 
-func checkArgs(_ *types.Event) error {
-	if len(handlerConfig.Example) == 0 {
-		return fmt.Errorf("--example or HANDLER_EXAMPLE environment variable is required")
+func checkArgs(event *corev2.Event) error {
+	if !event.HasMetrics() {
+		return fmt.Errorf("event does not contain metrics")
 	}
 	return nil
 }
 
-func executeHandler(event *types.Event) error {
-	log.Println("executing handler with --example", handlerConfig.Example)
+func executeHandler(event *corev2.Event) error {
+	if len(event.Metrics.Points) == 0 {
+		log.Println("event does not contain metric points")
+		return nil
+	}
+
+	proxyCfg := &wavefront.ProxyConfiguration{
+		Host:                 handlerConfig.Host,
+		MetricsPort:          handlerConfig.MetricsPort,
+		FlushIntervalSeconds: handlerConfig.FlushIntervalSeconds,
```

Comment:
Is this option useful if you're manually flushing (see the end of your function)?
I'm also expecting this handler to typically take much less than a second to run anyway?

Reply:
Wavefront establishes it's own default flush interval if this value is left unset anyway. I simply wanted to provide operators with a configuration option to flush quicker. For large payloads of metrics (which we've seen reported by some users already), I would expect this handler to run much longer than a second.